packages/component-library/src/components/overlay/mt-modal/mt-modal.vue

```diff
@@ -0,0 +1,205 @@
+  }
+});
+
+const modalRef = ref<HTMLElement | null>(null);
+let trap: ReturnType<typeof focusTrap.createFocusTrap> | undefined;
+
+watch(
+  isOpen,
+  async () => {
+    const isModalOpen = !!isOpen.value;
+    if (isModalOpen) await nextTick();
+
+    if (!modalRef.value) return;
+    if (isModalOpen) {
+      trap = focusTrap.createFocusTrap(modalRef.value, {
+        tabbableOptions: { displayCheck: "none" },
+        allowOutsideClick: true,
+      });
+
+      trap.activate();
+
+      return;
+    }
+
+    if (trap) trap.deactivate();
+  },
+  { immediate: true },
+);
+
+onUnmounted(() => {
+  if (trap) trap.deactivate();
+});
+</script>
+
+<style scoped lang="scss">
+.mt-modal {
+  height: auto;
+  position: fixed;
+  top: 50%;
+  left: 50%;
+  translate: -50% -50%;
+  z-index: 2000;
+  background-color: var(--color-elevation-surface-raised);
+  border-radius: 0.5rem;
+  overflow: hidden;
+  width: min(660px, calc(100vw - 2rem));
+  border: 1px solid var(--color-border-primary-default);
+  max-height: calc(100vh - 2rem);
+  display: flex;
+  flex-direction: column;
+}
+
+.modal-enter-active {
+  transition: opacity 400ms cubic-bezier(0.05, 0.7, 0.1, 1);
+
+  @media (prefers-reduced-motion: no-preference) {
+    transition:
+      opacity 400ms cubic-bezier(0.05, 0.7, 0.1, 1),
+      scale 400ms cubic-bezier(0.05, 0.7, 0.1, 1);
+  }
+}
+
+.modal-leave-active {
+  transition: opacity 200ms cubic-bezier(0.3, 0, 0.8, 0.15);
+
+  @media (prefers-reduced-motion: no-preference) {
+    transition:
+      opacity 200ms cubic-bezier(0.3, 0, 0.8, 0.15),
+      scale 200ms cubic-bezier(0.3, 0, 0.8, 0.15);
+  }
+}
+
+.modal-enter-from,
+.modal-leave-to {
+  opacity: 0;
+
+  @media (prefers-reduced-motion: no-preference) {
+    scale: 0.9;
+  }
+}
+
+.mt-modal__content {
+  --mt-modal-padding: 1.5rem;
+  --mt-modal-padding-bottom: 2.5rem;
+  --inset-top: var(--mt-modal-padding);
+  --inset-right: var(--mt-modal-padding);
+  --inset-bottom: var(--mt-modal-padding-bottom);
+  --inset-left: var(--mt-modal-padding);
+
+  padding: var(--mt-modal-padding);
+  padding-bottom: var(--mt-modal-padding-bottom);
+  overflow-y: auto;
+}
+
+.mt-modal__footer {
+  padding: 1.5rem;
+  background-color: var(--color-background-primary-disabled);
+}
+
+.mt-modal__header {
+  padding: 1.5rem;
+  border-bottom: 1px solid var(--color-border-primary-default);
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+}
+
+.mt-modal__title {
+  color: var(--color-text-primary-default);
+  margin-block-end: 0;
+  font-weight: $font-weight-medium;
+  line-height: $line-height-lg;
+  font-size: $font-size-m;
+}
+
+.mt-modal__header-content {
+  display: flex;
+  flex-direction: row;
+  align-items: baseline;
+  column-gap: 0.5rem;
+}
+
+.mt-modal__close-button {
+  cursor: pointer;
+  color: var(--color-icon-primary-default);
+  border-radius: $border-radius-default;
+  width: 2rem;
+  height: 2rem;
```

Comment:
```suggestion
  width: 2.5rem;
  height: 2.5rem;
```
Let's make this button a bit bigger to align to the recommended 40px minimum touch target size. Btw. 2.5rem only works because our default font-size (rem) seems to be 16px? Shouldn't it be 14px or do we keep them intentionally out of sync to have a rem size that's fitting in the 8px grid?

Reply:
Yes, I would leave it like this. If the token for the default font size will 14px it will be 0.875rem. The target size for WCAG AAA is 44px, it would then set it to 44px.